routes/actions.js

```diff
@@ -186,7 +186,7 @@ router.post('/signin', function(req, res, next) {
         debug.error({ req, error });
         return res.redirect('/signin');
       } else {
-        return res.redirect('/'); // Success
+        return res.redirect(req.body.returnto); // Success
```

Comment:
The tests are failing because the referer on the signin and register page is .. the signin and register page. Try logging in by going directly to localhost/signin and you'll see that you can't. :) But we shouldn't be using the referer at all, more on that below.